debugging/coreclr/dockerManager.ts

```diff
@@ -283,6 +276,61 @@ export class DefaultDockerManager implements DockerManager {
 
     private static readonly ProgramFilesEnvironmentVariable: string = 'ProgramFiles';
 
+    private static compareProperty<T, U>(obj1: T | undefined, obj2: T | undefined, getter: (obj: T) => (U | undefined)): boolean {
+        const prop1 = obj1 ? getter(obj1) : undefined;
+        const prop2 = obj2 ? getter(obj2) : undefined;
+
+        return prop1 === prop2;
+    }
+
+    private static compareDictionary<T>(obj1: T | undefined, obj2: T | undefined, getter: (obj: T) => ({ [key: string]: string } | undefined)): boolean {
+        const dict1 = (obj1 ? getter(obj1) : {}) || {};
+        const dict2 = (obj2 ? getter(obj2) : {}) || {};
+
+        const keys1 = Object.keys(dict1).sort();
+        const keys2 = Object.keys(dict2).sort();
+
+        if (keys1.length !== keys2.length) {
+            return false;
+        }
+
+        for (let i = 0; i < keys1.length; i++) {
+            if (keys1[i] !== keys2[i]) {
+                return false;
+            }
+
+            if (dict1[i] !== dict2[i]) {
```

Comment:
I don't understand how this check is working. How are you using an index on a dictionary instead of the key? Even if the index worked - wouldn't it be out of order since you sorted the keys but I don't think you sorted the dict?

Reply:
If this is actually a bug then I'm totally using it as more reason to use deep-equal :)

Reply:
Ach, you're right!  That's what I get for not writing tests for this comparison of options (which I've now done).